File: app/src/main/java/com/donald/musictheoryapp/Utils/Button/ColorButton.java
package com.donald.musictheoryapp.Utils.Button;

import android.content.Context;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffColorFilter;
import androidx.core.content.ContextCompat;

import com.donald.musictheoryapp.R;

public class ColorButton extends QuestionButton
{
	public ColorButton(Context context)
	{
		super(
			context,
			ContextCompat.getDrawable(context, R.drawable.background_question_button)
		);
	}

	public void setColor(int color)
	{
		background().setColorFilter(new PorterDuffColorFilter(color, PorterDuff.Mode.ADD));
	}
}

File: app/src/main/java/com/donald/musictheoryapp/customview/questionbutton/ColorButton.kt
package com.donald.musictheoryapp.customview.questionbutton

import android.content.Context
import androidx.core.content.ContextCompat
import com.donald.musictheoryapp.R
import android.graphics.PorterDuffColorFilter
import android.graphics.PorterDuff

class ColorButton(context: Context) : QuestionButton(
    context,
    ContextCompat.getDrawable(
        context, R.drawable.background_question_button
    ) ?: throw IllegalArgumentException("Unable to get background")
) {

    fun setColor(color: Int) {
        backgroundBase.colorFilter = PorterDuffColorFilter(color, PorterDuff.Mode.ADD)
    }

}